src/rest_vol.c

```diff
@@ -639,6 +647,58 @@ H5_rest_set_connection_information(void)
      * Attempt to pull in configuration/authentication information from
      * the environment.
      */
+    #ifdef RV_CURL_SOCKET
+    
+      /* This is just a placeholder URL for curl's syntax, its specific value is unimportant */
+      URL = "0";
+      URL_len = 1; 
+
+      if (NULL == (base_URL = (char *) RV_malloc(URL_len + 1)))
+        FUNC_GOTO_ERROR(H5E_VOL, H5E_CANTALLOC, FAIL, "can't allocate space necessary for placeholder base URL");
+    
+      strncpy(base_URL, URL, URL_len);
+      base_URL[URL_len] = '\0';
```

Comment:
I'm surprised this works without having a base_URL. I guess when using the domain sockets it's implicitly pointing to the HSDS root domain and then the server can accept URLs in the form of just "/home/test_user1" rather than something such as "https://hsds.hdf.test/home/test_user1/". @jreadey Do you know if this is the case?